/**:
  ros__parameters:
    adaptiveParams: true
    debugLog: true

    imu:
      use: false
      gravityAlign: true
      calibTime: 3
      bufferSize: 2000

    mapNode:
      publishFullMap: false
      publishFreq: 1.0
      leafSize: 0.25

    odomNode:
      lidarType: "livox" # livox  other
      lidarTopic: "livox/lidar"
      imuTopic: "livox/imu"
      # lidarTopic: "/os1_points"
      # imuTopic: "/imu/data_raw"

      odom_frame: odom
      child_frame: base_link

      initialPose:
        use: false
        position:
          x: 0.0
          y: 0.0
          z: 0.0
        orientation:
          w: 1.0
          x: 0.0
          y: 0.0
          z: 0.0

      preprocessing:
        cropBoxFilter:
          use: true
          size: 1.0
        voxelFilter:
          scan:
            use: true
            res: 0.5 #0.25
          submap:
            use: true
            res: 0.5

      keyframe:
        threshD: 5.0
        threshR: 45.0

      submap:
        keyframe:
          knn: 10
          kcv: 10
          kcc: 10

      gicp:
        minNumPoints: 10
        s2s:
          kCorrespondences: 10
          maxCorrespondenceDistance: 1.0
          maxIterations: 32  #32
          transformationEpsilon: 0.01
          euclideanFitnessEpsilon: 0.01
          ransac:
            iterations: 5
            outlierRejectionThresh: 1.0
        s2m:
          kCorrespondences: 20
          maxCorrespondenceDistance: 0.5
          maxIterations: 32
          transformationEpsilon: 0.01
          euclideanFitnessEpsilon: 0.01
          ransac:
            iterations: 5
            outlierRejectionThresh: 1.0
